icu4c/source/i18n/timezone.cpp

```diff
@@ -1591,6 +1591,21 @@ TimeZone::getCanonicalID(const UnicodeString& id, UnicodeString& canonicalID, UB
     return canonicalID;
 }
 
+UnicodeString&
+TimeZone::getIanaID(const UnicodeString& id, UnicodeString& ianaID, UErrorCode& status)
+{
+    ianaID.remove();
+    if (U_FAILURE(status)) {
+        return ianaID;
+    }
+    if (id.compare(ConstChar16Ptr(UNKNOWN_ZONE_ID), UNKNOWN_ZONE_ID_LENGTH) == 0) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
```

Comment:
ZoneMeta::getIanaID will set the returned ianaID to bogus, should we also set ianaID to bogus here for this error?

Reply:
I think that is in the category of a "nice to have" consistency fix, but maybe it can be fixed later, should not block this PR.

Reply:
Thanks for pointing out. I'll make this change.